examples/api/lib/material/snack_bar/snack_bar.2.dart

```diff
@@ -0,0 +1,102 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/// Flutter code sample for [SnackBar] with Material 3 specifications.
+
+import 'package:flutter/material.dart';
+
+void main() => runApp(const MyApp());
+
+// A Material 3 [SnackBar] demonstrating an optional icon, in either floating
+// or fixed format.
+class MyApp extends StatelessWidget {
+  const MyApp({super.key});
+
+  static const String _title = 'Flutter Code Sample';
+
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      title: _title,
+      theme: ThemeData(
+      useMaterial3: true,
+      ),
+      home: Scaffold(
+        appBar: AppBar(title: const Text(_title)),
+        body: const Center(
+          child: SnackBarExample(),
+        ),
+      ),
+    );
+  }
+}
+
+enum SnackBarType { fixed, floating }
```

Comment:
Is this used anywhere? I see `SnackBarBehavior` used below, but I don't see `SnackBarType`.